system/Email/Email.php

```diff
@@ -2019,13 +2012,15 @@ protected function SMTPAuthenticate()
             return true;
         }
 
-        if ($this->SMTPUser === '' && $this->SMTPPass === '') {
+        // If no username or password is set
+        if ($this->SMTPUser === '' || $this->SMTPPass === '') {
             $this->setErrorMessage(lang('Email.noSMTPAuth'));
 
             return false;
         }
 
-        $this->sendData('AUTH LOGIN');
+        // send initial 'handshake' command
+        $this->sendData('AUTH ' . strtoupper($this->SMTPAuthMethod));
```

Comment:
If we rely on the `SMTPAuthMethod` to set the value of the `AUTH` command, then we should validate it first.

We should either hardcode `AUTH` for each method or move validation of the method before the switch statement, so we can return error message: `Email.noSMTPAuthMethod`.